locator/topology.cc

```diff
@@ -307,6 +312,10 @@ void topology::index_node(const node* node) {
     _dc_endpoints[dc].insert(endpoint);
     _dc_racks[dc][rack].insert(endpoint);
     _datacenters.insert(dc);
+
+    if (node->is_this_node()) {
+        _this_node = node;
+    }
```

Comment:
nit: please add a comment that the function must no throw below this point

Reply:
It actually can throw, when invoked from add_node(), the node will be popped, which will revert this change.

When called from update_node(), failure is not handled, but that seems to be a more general bug in update_node(), because index_node() may throw and partially index the node. What are the exception safety guarantees of update_node()? I don't see it documented.

Reply:
That's a good point.
You're right about reverting the change in pop_node and about mutating a temporary copy at the higher level.